implementation/src/main/java/io/smallrye/graphql/execution/LRUCache.java

```diff
@@ -0,0 +1,98 @@
+package io.smallrye.graphql.execution;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class LRUCache<K, V> {
+
+    private final int maxSize;
+    private final Map<K, Entry<V>> cache = new HashMap<>();
+    private Entry<V> start;
+    private Entry<V> end;
+
+    LRUCache(int maxSize) {
+        this.maxSize = maxSize;
+    }
+
+    V get(K key) {
+        Entry<V> entry = cache.computeIfPresent(key, this::moveEntryToStart);
+        return entry == null ? null : entry.value;
+    }
+
+    V put(K key, V value) {
+        Entry<V> entry = cache.get(key);
+        V oldValue = null;
+        if (entry == null) {
+            entry = cachePutNew(key, value);
+        } else {
+            entry = removeEntry(entry);
+            oldValue = entry.value;
+            entry.value = value;
+        }
+        entry = addToStart(entry);
+        return oldValue;
+    }
+
+    V computeIfAbsent(K key, Function<K, V> valueFunction) {
+        Entry<V> entry = cache.get(key);
+        if (entry == null) {
+            entry = cachePutNew(key, valueFunction.apply(key));
+        } else {
+            entry = removeEntry(entry);
+        }
+        entry = addToStart(entry);
+        return entry.value;
+    }
+
+    private Entry<V> cachePutNew(K key, V value) {
```

Comment:
Shouldn't these methods be synchronized like the others, because `HashMap#get`, `#put` and `#remove` aren't threadsafe?

Reply:
I've made `cache` to be a ConcurrentHashMap, and put a sync block around the remove case.  I think the rest can be left un-sync'd for better performance.  Wdyt?

Reply:
lgtm :)